### PR TITLE
Move the stage0 data segment to low memory

### PR DESCRIPTION
**Pull request by conradgrobler**
This moves the target location DATA segment into low memory, while still storing the content in the ROM image. The bootstrap assembly then copies it from the ROM image into the expected memory location in low memory.

### CODE REVIEW

File: stage0/layout.ld [hunk 1]
```diff
@@ -75,6 +75,13 @@ SECTIONS {
         bss_size = . - bss_start;
     } > ram_low
 
+    .data : 
+    AT ( ADDR (.text) + SIZEOF (.text) ) {
```
Comment (andrisaar) [inline stage0/layout.ld:79]: Could you do `AT(text_end)` or something similar? Or I guess `text_end` might not have a value at this point yet...

All in all, this feels like it'll be quite fragile, but I don't know of a better way how to do it from the top of my head either.

**Review by andrisaar — APPROVED** [review]
Just to double-check, could you dump the section headers from `readelf` to show that all the sections have the expected LMAs and VMAs?

File: stage0/layout.ld (hunk 1 above)
Comment (conradgrobler) [reply stage0/layout.ld:79]: I could not find a better way that worked yet. I agree that it is a bit fragile. If I think of a better way I will fix it in a future PR.

**Comment by conradgrobler** [comment]
> Just to double-check, could you dump the section headers from `readelf` to show that all the sections have the expected LMAs and VMAs?

```
  [Nr] Name              Type            Address          Off    Size   ES Flg Lk Inf Al
  [ 3] .data             PROGBITS        000000000002b010 001010 000010 00  WA  0   0  8

Program Headers:
  Type           Offset   VirtAddr           PhysAddr           FileSiz  MemSiz   Flg Align
  LOAD           0x001000 0x0000000000001000 0x0000000000001000 0x000000 0x002000 R   0x1000
  LOAD           0x001000 0x0000000000003000 0x0000000000003000 0x000000 0x028010 RW  0x1000
  LOAD           0x001010 0x000000000002b010 0x00000000ffff87b5 0x000010 0x074ff0 RW  0x1000
  LOAD           0x002000 0x00000000fffe0000 0x00000000fffe0000 0x0073d8 0x0073d8 R   0x1000
  LOAD           0x0093e0 0x00000000fffe73e0 0x00000000fffe73e0 0x018c20 0x018c20 R E 0x1000
  GNU_STACK      0x000000 0x0000000000000000 0x0000000000000000 0x000000 0x000000 RW  0

 Section to Segment mapping:
  Segment Sections...
   02     .data .stack .ebda 
```

So ,data maps to the segment with virtual address 0x000000000002b010 and physical address 0x00000000ffff87b5 and has a size in the file of 16 bytes, which is also the size of the data section.

**Pull request merged**